java-util/src/main/java/io/druid/java/util/http/client/NettyHttpClient.java

```diff
@@ -280,18 +280,18 @@ public void exceptionCaught(ChannelHandlerContext context, ExceptionEvent event)
             if (response != null) {
               handler.exceptionCaught(response, event.getCause());
             }
-            removeHandlers();
             try {
-              channel.close();
+              if (channel.isOpen()) {
+                channel.close();
+              }
             }
             catch (Exception e) {
-              // ignore
+              log.warn(e, "Error while closing channel");
             }
             finally {
+              removeHandlers();
```

Comment:
Hmm, this causes another wacky error like below:

```
2018-08-03T21:17:17,413 WARN [HttpClient-Netty-Worker-0] org.jboss.netty.channel.DefaultChannelPipeline - An exception was thrown by a user handler while handling an exception event ([id: 0x2a95abde, /127.0.0.1:58801 :> localhost/127.0.0.1:58798] EXCEPTION: java.lang.IllegalArgumentException: invalid version format: GET)
java.util.NoSuchElementException: last-handler
```

This is because the channel is already closed when we try to remove handlers. Let's just remove this line. It should be fine because the channel is already closed.

Reply:
Removed.